tests/interpretation/test_linear_probe.py

```diff
@@ -149,6 +163,19 @@ def test_evaluate_probe_with_class_labels(self):
         self.assertIn("class0", scores)
         self.assertIn("class1", scores)
 
+    def test_evaluate_probe_with_class_labels_float16(self):
```

Comment:
Can you make sure that these new tests cover all the new code that was added? I think they do but it would be great if you can confirm; You can run

```bash
python -m pytest -vv --cov --cov-report term-missing
```

to see the exact lines that are not covered by the test. I'll make this the default behavior for the future so that `run_tests` shows this information

Reply:
Regarding evaluation: 
I played a bit with `python -m timeit` and dtype conversion on an average CPU, and a GPU. I found that the conversion from float16 to float32 does not take much time, maybe 100-200ms for converting BERT base activations for 1000 ten-word-sentences. This is barely noticeable in practice, so it's a good argument against the special case that is in the code now.

That being said, you need some additional code now because you need to make sure that X and the probe have the same dtype, and that the dtype is not float16 if you're on CPU. I suggest doing that by always enforcing evaluation in float32, regardless of the input formats. I pushed a simpler version of the eval method in commit 4cdc823d74aba55096155246435b53ac92e47a1d . Do you like it better as well?